examples/parcel/src/ui/css-map.jsx

```diff
@@ -0,0 +1,12 @@
+import { css, cssMap } from '@compiled/react';
+
+const styles = cssMap({
+  danger: {
+    color: 'red',
+  },
+  success: {
+    color: 'green',
+  },
+});
+
+export default ({ variant, children }) => <div css={css(styles[variant])}>{children}</div>;
```

Comment:
Just walking my way through this...

1. Just using javascript to choose the style
```js
const styles = css.create({
    danger: {
       color: 'red'
    },
    success: {
      color: 'green'
    }
});

const map = {
  danger: styles.danger,
  success: styles.success
}

export default ({variant, children}) => <div css={css(map[variant])} >{children}</div>
```
Can compile to
```js
const styles = {
    danger: {
       color: '<hash_color_red>'
    },
    success: {
      color: '<hash_color_green>'
    }
});

const map = {
  danger: styles.danger, // == {color: <hash_color_red>}
  success: styles.success // == {color: <hash_color_green>}
}

export default ({variant, children}) => <div css={css(map[variant])} >{children}</div>
```

2. Consumer passes in a _style_ itself
```js
export default ({ extendedStyle, children }) => <div css={css(extendedStyle)}>{children}</div>;
```
can compile to
```js
export default ({ extendedStyle, children }) => <div css={css(extendedStyle)}>{children}</div>;
```

You're combining those two cases together... to form
3. 
```js
const styles = css.match({
    danger: {
       color: 'red'
    },
    success: {
      color: 'green'
    }
});
export default ({ variant, children }) => <div css={css(styles[variant])}>{children}</div>;
```
seems like a nice shortcut to me, but I'm unsure if we need it to be a _different and special_ construct vs what we could make possible with 1

Reply:
Compiled essentially transforms one CSS rule into three pieces.
1. classNames
2. sheets
3. CSS variables (if there are variables cannot be resolved statically). For simplicity, let's ignore it.

We can conditionally apply classNames to DOM element, but sheets cannot. Hence a special construct help identifying variants (will be useful for lints too), and piece sheets together. 

```
const styles = cssMap({
    danger: {
       color: 'red'
    },
    success: {
      color: 'green'
    }
});
export default ({ variant }) => <div css={css(styles[variant])} />;
``` 

gets compiled to

```
<CS>".a {color: red}", ".b {color: green}"</CS> // sheets need to include all possible CSS outcome
...
```

I think the approach you mentioned from 1 is more leaning on local tranforms (which is roadmapped 🙂). If we have local transforms:
```
const foo = css({ 
    color: 'red' 
})

const bar = cssMap({
    danger: {
       color: 'red'
    },
    success: {
      color: 'green'
    }
});
```
gets compiled to
```
const foo = {
  className: ".a",
  sheet: ".a{color:red}",
};

const bar = {
    danger: {
         className: ".a",
         sheet: ".a{color:red}",
    },
    success: {
         className: ".b",
         sheet: ".b{color:green}",
    }
}

```
We can easily use javascript to choose the style
```
<div css={css([foo, bar[variant]])} />
```
gets compiled to
```
<CS>{[foo, bar]}</CS>
<div className={ax(foo.className, bar[variant].className])} />
```
If I misunderstood the comment, let's have a quick zoom call to clarify it (maybe)?